plansys2_terminal/src/plansys2_terminal/Terminal.cpp

```diff
@@ -891,6 +954,113 @@ Terminal::process_check(std::vector<std::string> & command, std::ostringstream &
   // run (move leia entrance dinning)
 }
 
+void
+Terminal::process_help(std::vector<std::string> & command, std::ostringstream & os)
+{
+  std::ostringstream cmd_set;
+  cmd_set << "\t set instance <object> <type>" << std::endl <<
+    "\t set predicate <predicate>" << std::endl <<
+    "\t set function (= <function> <value>)" << std::endl <<
+    "\t set goal <goal_formula>" << std::endl;
+  std::ostringstream cmd_get;
+  cmd_get << "\t get model   types | predicates | functions | actions |" << std::endl <<
+    "\t             predicate <predicate_name> | function <function_name> |" << std::endl <<
+    "\t             action <action_name> " << std::endl <<
+    "\t get problem  instances | predicates | functions | goal " << std::endl <<
+    "\t get domain" << std::endl <<
+    "\t get plan" << std::endl;
+  std::ostringstream cmd_rem;
+  cmd_rem << "\t remove instance <object>" << std::endl <<
+    "\t remove predicate <predicate>" << std::endl <<
+    "\t remove function <function>" << std::endl <<
+    "\t remove goal" << std::endl;
+
+  std::ostringstream cmd_run;
+  cmd_run << "\t run action <action>" << std::endl <<
+    "\t run num_actions <num>" << std::endl <<
+    "\t run planfrom <planfile>" << std::endl;
+
+  std::ostringstream cmd_check;
+  cmd_check << "\t check [actors]" << std::endl;
+
+  std::ostringstream cmd_source;
+  cmd_source << "\t source <file_name> [0|1]" << std::endl;
+
+  std::ostringstream cmd_help;
+  cmd_help << "\t help|? [ set | get | remove | run | check |" <<
+    " source | quit | help | ? ]" << std::endl;
+
+  std::map<std::string, std::string> cmds;
+  cmds.insert(std::pair<std::string, std::string>("set", cmd_set.str()));
+  cmds.insert(std::pair<std::string, std::string>("get", cmd_get.str()));
+  cmds.insert(std::pair<std::string, std::string>("remove", cmd_rem.str()));
+  cmds.insert(std::pair<std::string, std::string>("run", cmd_run.str()));
+  cmds.insert(std::pair<std::string, std::string>("check", cmd_run.str()));
+  cmds.insert(std::pair<std::string, std::string>("source", cmd_source.str()));
+  cmds.insert(std::pair<std::string, std::string>("quit", std::string("")));
+  cmds.insert(std::pair<std::string, std::string>("help", cmd_help.str()));
+  cmds.insert(std::pair<std::string, std::string>("?", cmd_help.str()));
+
+  if (command.empty()) {
+    os << "The list of available commands is the following:" << std::endl;
+    for (auto it1 = cmds.begin(); it1 != cmds.end(); it1++) {
+      os << it1->first << std::endl << it1->second << std::endl;
+    }
+    return;
+  }
+
+  const std::string c = command[0];
+  auto cmd = cmds.find(c);
+  if (cmd != cmds.end()) {
+    os << cmd->first << std::endl;
+    os << cmd->second << std::endl;
+    return;
+  }
+  os << "Command \"" << command[0] << "\" command not found" << std::endl;
+}
+
+void
+Terminal::process_source(std::vector<std::string> & command, std::ostringstream & os)
```

Comment:
I would prefer to avoid this command. I still think that the effect of this command is similar to opening another terminal and run  `ros2 run plansys2_terminal plansys2_terminal < file_with_commands`. This part only adds complexity to the terminal, with variables set inside and outside the function, prone to lateral effects.

Reply:
You may simulate this with yet another terminal, fine but if you're interacting over the terminal, and you would like to continue interacting with the terminal, instead of opening another terminal, set up the ros envirionemt, and run the command, you can do the very same thing from one single terminal.

To be honest, playing with the terminal I found this feature very useful! We can think to a better mechanism instead of variables, but I strongly believe that having this feature is useful and the "complexity" added is manageable.

Reply:
Ok, let's add his feature to the terminal, but we have to find a better implementation that does not use global variables, like `inside_source` and `finish_parsing`

Reply:
I'm trying an implementation without these global variables, and I'll commit soon as all the test passes

Reply:
I've made the code reentrant by removing global variables.

All tests pass.